File: sort_tool.py
from pathlib import Path
from check_file_permission import check_file_permission

def sort_file(file_path, config, reverse=False):
    if not check_file_permission(file_path, "write", config):
        raise PermissionError(f"Write access denied for {file_path}")
    file_path = Path(file_path).resolve()
    if not file_path.is_file():
        raise ValueError(f"File not found: {file_path}")
    with open(file_path, "r", encoding="utf-8") as f:
        lines = f.readlines()
    lines.sort(reverse=reverse)
    with open(file_path, "w", encoding="utf-8") as f:
        f.writelines(lines)
    return f"Sorted {file_path}"

File: ts-tools/sort_tool.ts
/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function sortFile(
  filePath: string,
  config: any,
  reverse: boolean = false,
): Promise<string> {
  return new Promise((resolve, reject) => {
    checkFilePermission(filePath, 'write', config)
      .then(() => {
        const resolvedPath = path.resolve(filePath);
        fs.readFile(resolvedPath, 'utf-8', (err, data) => {
          if (err) {
            reject(err);
          } else {
            const lines = data.split('\n');
            lines.sort();
            if (reverse) {
              lines.reverse();
            }
            fs.writeFile(resolvedPath, lines.join('\n'), 'utf-8', (err) => {
              if (err) {
                reject(err);
              } else {
                resolve(`Sorted ${filePath}`);
              }
            });
          }
        });
      })
      .catch(reject);
  });
}
